src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -0,0 +1,601 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *
+ *    Copyright (c) 2020 Silicon Labs
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance  the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "window-covering-server.h"
+
+#include <app/util/af.h>
+
+#include <app/common/gen/attribute-id.h>
+#include <app/common/gen/cluster-id.h>
+#include <app/common/gen/command-id.h>
+
+#include <app/CommandHandler.h>
+#include <app/common/gen/attributes/Accessors.h>
+#include <app/reporting/reporting.h>
+#include <app/util/af-event.h>
+#include <app/util/af-types.h>
+#include <app/util/attribute-storage.h>
+#include <string.h>
+
+#ifdef EMBER_AF_PLUGIN_SCENES
+#include <app/clusters/scenes/scenes.h>
+#endif // EMBER_AF_PLUGIN_SCENES
+
+using namespace chip::app::Clusters;
+
+#define WC_PERCENT100THS_MAX 10000
+
+//------------------------------------------------------------------------------
+// WindowCover
+//------------------------------------------------------------------------------
+
+WindowCover WindowCover::sInstance;
+
+WindowCover & WindowCover::Instance()
+{
+    return sInstance;
+}
+
+WindowCover::WindowCover() {}
+
+void WindowCover::TypeSet(EmberAfWcType type)
+{
+    std::underlying_type<EmberAfWcType>::type value = static_cast<std::underlying_type<EmberAfWcType>::type>(type);
```

Comment:
There's still value in doing the simpler `to_underlying` bit here while we sort out how the accessors are generated...

Reply:
chip::to_underlying used instead.